cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml

```diff
@@ -667,6 +677,213 @@ Resources:
       MinSize: !Ref AsgWarmPoolMinSize
       PoolState: !Ref AsgWarmPoolState
 
+  # ASG Lambda Related resources
+
+  CcCloudwatchLogGroupForLambda:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub '${AWS::StackName}-CcCloudwatchLogGroupForLambda'
+
+  CcGetSecretsPolicyForLambda:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Zscaler Cloud Connector Get Secrets Policy For Lambda
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'secretsmanager:GetSecretValue'
+
+  CcCloudwatchMetricsPolicyForLambda:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Zscaler Cloud Connector Cloudwatch Metrics Policy For Lambda
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'cloudwatch:GetMetricStatistics'
+              - 'cloudwatch:ListMetrics'
+              - 'cloudwatch:PutMetricData'
+              - 'ec2:DescribeTags'
+            Resource:
+              - '*'
+
+  CcCloudwatchLogsPolicyForLambda:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Zscaler Cloud Connector Cloudwatch Logs Policy For Lambda
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'logs:CreateLogStream'
+              - 'logs:PutLogEvents'
+            Resource:
+              - !GetAtt 
+                - CcCloudwatchLogGroupForLambda
+                - Arn
+          - Effect: Allow
+            Action:
+              - 'logs:CreateLogGroup'
```

Comment:
@anichari take a look at this commit. After talking with venkatesh, I ended up dropping the CreateLogGroup IAM in lieu of a separate Log group creation in order to set a custom 14 day retention time.

https://github.com/zscaler/terraform-aws-cloud-connector-modules/commit/ab2e1dd6fdfcda7509b4f114f2e7a2d4f0e8f7f4#diff-ba0f81d84e66eac04b6f0f063d0c824ee174c680c697e8d451c125ab60b83f37